### PR TITLE
Replace deprecated `cgi` with `email` from standard library

### LINKED ISSUE #71: `cgi` is being deprecated in 3.13
https://peps.python.org/pep-0594/#cgi

The official docs suggest to replace `cgi.parse_header` with `email.message.Message`:

> As an explicit example of how close parse_header and email.message.Message are:
> ```
> >>> from cgi import parse_header
> >>> from email.message import Message
> >>> parse_header(h)
> ('application/json', {'charset': 'utf8'})
> >>> m = Message()
> >>> m['content-type'] = h
> >>> m.get_params()
> [('application/json', ''), ('charset', 'utf8')]
> >>> m.get_param('charset')
> 'utf8'
> ```

@siddhantgoel I'm happy to take a shot at the PR if you're accepting them!

### PR DESCRIPTION
**Pull request by jasopolis**
Starting from Python 3.11, [`cgi` is deprecated](https://docs.python.org/3/library/cgi.html) and emits warnings; the official docs suggest to [use `email` instead](https://peps.python.org/pep-0594/#cgi). 

So here's an updated implementation doing exactly that -- all tests pass locally for me, and `make speed-test` shows no noticeable differences.

Thanks in advance @siddhantgoel !

Closes #71

### CODE REVIEW

File: streaming_form_data/parser.py [hunk 1]
```diff
@@ -20,12 +20,13 @@ def parse_content_boundary(headers: Mapping[str, str]) -> bytes:
     if not content_type:
         raise ParseFailedException('Missing Content-Type header')
 
-    value, params = cgi.parse_header(content_type)
+    m = EmailMessage()
```
Comment (siddhantgoel) [inline streaming_form_data/parser.py:23]: Let's make this variable name a bit more descriptive. Perhaps something like `message`?

File: streaming_form_data/_parser.pyx [hunk 2]
```diff
@@ -275,27 +276,24 @@ cdef class _Parser:
                 if byte != c_lf:
                     self.mark_error()
                     return ErrorGroup.PartHeaders + 1
+                
+                m = Parser(policy=HTTP).parsestr(chunk[buffer_start: idx + 1].decode('utf-8'))
```
Comment (siddhantgoel) [inline streaming_form_data/_parser.pyx:280]: Let's make this variable name a bit more descriptive. Perhaps something like `message`?

File: README.md [hunk 3]
```diff
@@ -67,8 +66,14 @@ compiler available. On Debian-based distros this usually means installing the
 2. Install the packages required for development:
    `make pip-sync`
 
-3. That's basically it. You should now be able to run the test suite:
-   `make test`.
+3. Compile the C extensions in `streaming_form_data`:
+   `make compile`
```
Comment (siddhantgoel) [inline README.md:70]: The C file is also checked in as part of the source code so this step wouldn't really change anything. I wonder if there's a different reason you added this step here? 🤔

**Review by siddhantgoel — COMMENTED** [review]
Thanks for the PR! Overall everything looks good, just left some minor edit suggestions.

Also, I just merged your other PR that adds Python 3.11 support. Could you rebase this branch from `main` so the CI also runs tests against 3.11 on this branch?

After those two things are addressed we could merge this in!

File: README.md (hunk 3 above)
Comment (jasopolis) [reply README.md]: You're right, I forgot about that, sorry! I just wanted to clarify the instructions for those without much Cython experience (like me 😅). I reworded it as a separate note in 5a78097, but I'm also happy to take it out altogether if you prefer.

File: streaming_form_data/parser.py (hunk 1 above)
Comment (jasopolis) [reply streaming_form_data/parser.py]: Done in 188232e, thanks!

File: streaming_form_data/_parser.pyx (hunk 2 above)
Comment (jasopolis) [reply streaming_form_data/_parser.pyx]: Also done in 188232e

**Comment by jasopolis** [comment]
(Also, apologies for the force push -- realized I had the wrong email on the commits!)

**Review by siddhantgoel — APPROVED** [review]

**Comment by siddhantgoel** [comment]
No worries. I've merged this in and will try to publish the next release over the weekend!

**Comment by jasopolis** [comment]
Awesome! Thanks -- really appreciate that you're so actively maintaining this library :)

**Comment by siddhantgoel** [comment]
Alright, just published v1.11.1 on PyPI which includes this!

**Pull request merged**